libethereum/Executive.cpp

```diff
@@ -373,6 +395,9 @@ bool Executive::go(OnOpFunc const& _onOp)
 			auto vm = _onOp ? VMFactory::create(VMKind::Interpreter) : VMFactory::create();
 			if (m_isCreation)
 			{
+				if (m_envInfo.number() >= m_sealEngine.chainParams().u256Param("metropolisForkBlock") && m_s.addressHasCode(m_newAddress))
```

Comment:
This will affect both creation transaction and CREATE2, but this rule is specified only for CREATE2 in EIP. I've asked for the clarification there.